services/sonar/sonar-tech-debt.service.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const SonarBase = require('./sonar-base')
+const {
+  patternBase,
+  queryParamSchema,
+  getLabel,
+  badMetricColorScale,
+} = require('./sonar-helpers')
+
+module.exports = class SonarTechDebt extends SonarBase {
+  static get category() {
+    return 'analysis'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'tech debt' }
+  }
+
+  static render({ debt, metric }) {
+    return {
+      label: getLabel({ metric }),
+      message: `${debt}%`,
+      color: badMetricColorScale(debt),
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'sonar',
+      pattern: `${patternBase}/:metric(tech_debt|sqale_debt_ratio)`,
+      queryParamSchema,
+    }
+  }
+
+  async handle({ protocol, host, component, metric }, { version }) {
+    const json = await this.fetch({
+      version,
+      protocol,
+      host,
+      component,
+      //special condition for backwards compatibility
+      metricName: 'sqale_debt_ratio',
```

Comment:
This was replicated from the legacy service implementation, but I'm not 100% convinced it is needed in all cases (at least not for newer versions of Sonar). Need to do some more reading on this

Reply:
Leaving this as-is for now